ceph_iscsi_config/client.py

```diff
@@ -181,7 +182,8 @@ def define_client(self):
 
         # NB. this will check all tpg's for a matching iqn
         for client in r.node_acls:
-            if client.node_wwn == self.iqn:
+            target = client.parent_tpg.parent_target
+            if client.node_wwn == self.iqn and target.wwn == self.target_iqn:
```

Comment:
Instead of looping over all the clients, you can just limit it to the specific target since you added it above. Something like

```
target = Target(iscsi, self.target_iqn, 'lookup')
for each tpg in target.tpgs:
     for client in tpg.node_acls:
```

Reply:
Looks better now, thanks. Fixed.